src/casp/cell_data_manager/README.md

```diff
@@ -0,0 +1,118 @@
+    project="test-project",
+    dataset="my-test-dataset",
+    select=select_columns,
+   **additional_kwargs
+)
+```
+3. Render SQL Query
+```python3
+rendered_sql_query = sql.render(
+    template_path="your_path_to_templates/templated_query.sql.mako",
+    template_data=template_data,
+)
+# Execute your query...
+```
+## Module Structure
+
+The Cell Data Manager `cell_data_manager` module consists of the following structure:
+
+```
+├── data_controller # (To be developed): This should be developed during `bq_scripts` refactor
+└── sql
+    ├── __init__.py
+    ├── constants.py
+    ├── mako_helpers.py  # Helper functions used in SQL templates
+    ├── query.py  # Query management / validation directory
+    ├── template_data.py # Template Data Parsing class
+    └── templates # All SQL templates go here
+        └── ... 
+    └── validation
+        └── ...                
+```
+## Security Considerations
+
+### SQL Injection Awareness
+
+When working with SQL Template building, it's important to be aware of SQL injection vulnerabilities. SQL Template building relies on string interpolation, which can be susceptible to SQL injection attacks. Current implementation of SQL query construction doesn't leverage SQL injection checks.
+
+**Scope**: All SQL queries executed within `bq_scripts` are designed exclusively for internal use within the `cellarium-cloud` infrastructure. `bq_scripts` is a secure module and has never been exposed to user clients or APIs. Therefore, it is safe to use template building techniques for managing our internal SQL queries.
+
+### Best Practices
+
+To ensure the security of your application, please follow these best practices:
+
+1. **Avoid External Exposure**: Do not use SQL template building tools for constructing queries that will be exposed to end-users or external entities.
+
+2. **Parameterized Queries**: If there is a need to expose SQL queries to end-users, always use [Parameterized Queries](https://cloud.google.com/bigquery/docs/parameterized-queries#python). Parameterized queries help prevent SQL injection by separating data from SQL code.
+
+3. **Consider extending `mako_helpers`**: To simplify the implementation of parameterized queries, consider expanding `mako_helpers` to work seamlessly with [Parameterized Queries](https://cloud.google.com/bigquery/docs/parameterized-queries#python).
+
+By following these practices, you can ensure the safe and secure use of this module within the project.
+
+## To Consider
+
+Currently, there's no SQL formatting in the `casp.cell_data_manager.sql` module. There's no SQL formatter tool that supports the Mako template manager. However, there is one that supports Jinja:
+(sqlfmt)[https://github.com/tconbeer/sqlfmt]. I didn't want to use Jinja just because there's a formatting tool that supports it. Hopefully, we will find a SQL formatting tool suitable for our use case, or we might consider contributing to this repository to write a Mako formatting connector.
```

Comment:
I -> We.

I suggest reading this README from top to bottom and making it a bit more formal. Thanks!

Reply:
https://github.com/cellarium-ai/cellarium-cloud/issues/108 

I'll work on this pretty soon.